### DISCUSSION TITLE
SPA with Quasar-Cli (2.17.7), Vue 3 and Webpack - Problem with CSP

### DISCUSSION THREAD

**Discussion by Shocking2709** [quasarframework/quasar#16375] [General - Components / Directives / etc]
I'm working on a SPA with quasar 2.17.7 (quasar-cli and webpack) and now have the task of setting the CSP header (content-security-policy). The use of unsafe-inline and unsafe-eval is explicitly prohibited.

The solution I found online was to use vue.runtime.js instead of the full compiler version. I thought this was already given by `"vueCompiler: false"` in quasar.config.js. Is that correct?

In addition, precompiled templates would have to be used.  To be honest, I have no idea how to configure Webpack to do this. I found some hints on using css-loader and style-loader, but I did not know how to configure. I tried the following in quasar.config.js:

```
chainWebpack(chain) {
                chain.plugin('eslint-webpack-plugin')
                    .use(ESLintPlugin, [{extensions: ['js', 'vue']}])
                chain.module
                    .rule('i18n-resource')
                    .test(/\.(json5?|ya?ml)$/)
                    .include.add(path.resolve(__dirname, './src/i18n'))
                    .end()
                    .type('javascript/auto')
                    .use('i18n-resource')
                    .loader('@intlify/vue-i18n-loader')
                chain.module
                    .rule('i18n')
                    .resourceQuery(/blockType=i18n/)
                    .type('javascript/auto')
                    .use('i18n')
                    .loader('@intlify/vue-i18n-loader')
              chain.module
                  .rule('css-loader').test('/\.css$/i')
                  .use('css-loader').loader('css-loader')
              chain.module
                .rule('style-loader').test('/\.css$/i')
                .use('style-loader').loader('style-loader')
            }
```
But this did not work at all. Where is the option to configure the precompile of js-files?

And are these actually the right ways to avoid having to use ansafe-eval and unsafe-inline? Or is there a better way to get a csp-save application with vue 3?

**Comment by Shocking2709** [comment]
Is here really nobody who can answer my question? Or is the question not understandable?

**Comment by EthraZa** [comment]
Just searching for this subject to evaluate Quasar for a next project. 
Any updates?

**Reply by rstoenescu** [reply]
Use q/app-vite instead of q/app-webpack. CSP is iffy with webpack (not Quasar‘s fault)
